src/domains/User/UserProfile/UserProfile.tsx

```diff
@@ -19,37 +31,185 @@ const useStyles = makeStyles((theme) => ({
     },
 }));
 
+const RequiredUser = () => {
+    const [user, setUser] = useUser();
+    // if no user is logged in
+    if (!user) throw new Error('RequiredUser() should never be accessed in a context where the user may not be logged in');
+    return [user, setUser] as const;
+}
+
 export default function UserProfile({ img }: Props) {
     const classes = useStyles();
+    const [user, setUser] = RequiredUser();
+    const initState: RegisterForm = { firstName: user.name.first, lastName: user.name.last, email: user.email.address, password: '', confirmPassword: '' };
```

Comment:
I think putting this outside the scope of the `UserProfile` function (so before line 41) makes sense for a few reasons:
1.  This code reruns every render (so this variable gets "remade" every rerender)
2. It isn't dependent on the react life cycle at all since it's just an initial state.  Notice how everything else that you're declaring is either a custom hook or a `React.use` hook -- ie it's plugged into React's lifecycle.
3. Personally, I think putting the initial state is more readable, but it's really a matter of personal preference.

Reply:
Done, moved to outside the function.